scripts/release-notes-jira.mjs

```diff
@@ -0,0 +1,113 @@
+/**
+ * Generates release notes from completed Jira stories and bugs.
+ * 
+ * Required command-line parameters:
+ * - Jira project key (e.g., "LARA")
+ * - Jira label for filtering stories (e.g., "lara-5.0.0")
+ * 
+ * Optional command-line parameters:
+ * - "slack" to format output for Slack
+ * 
+ * Example usage:
+ * node release-notes-jira.mjs LARA lara-5.0.0 slack
+ */
+
+import "dotenv/config"
+import fetch from "node-fetch";
+import querystring from "querystring";
+import { extractBlurbText, jiraApiBaseUrl, jiraRequestHeaders } from "./utils.mjs";
+
+const jiraUser = process.env.JIRA_USER;
+const jiraToken = process.env.JIRA_TOKEN;
+
+if (!jiraUser || !jiraToken) {
+  console.error("Both the JIRA_USER and JIRA_TOKEN environment variables are required.");
+  process.exit(1);
+}
+
+const features = [];
+const bugs = [];
+const underTheHood = [];
+
+function isUnderTheHood(story) {
+  return story.fields.labels.find(label => label === "under-the-hood");
+}
+
+async function collectStories(projectKey, search) {
+  const fields = ["id", "summary", "issuetype", "description", "labels"].join(",");
+  const jql = `project=${projectKey} AND issuetype in (Story, Bug) AND status in (Done, Closed) AND labels in (${search})`;
+  const urlQuery = querystring.stringify({
+    jql,
+    fields,
+    maxResults: 100
+  });
+  const url = `${jiraApiBaseUrl}/search?${urlQuery}`;
+  const requestHeaders = jiraRequestHeaders(jiraUser, jiraToken);
+  const response = await fetch(url, requestHeaders);
+  const json = await response.json();
+  const stories = json.issues;
+
+  if (stories.length === 0) {
+    console.error(`No stories found for project ${projectKey} with label ${search}`);
+    process.exit(1);
+  }
+
+  for (const story of stories) {
+    if (isUnderTheHood(story)) {
+      underTheHood.push(story);
+    } else {
+      if (story.fields.issuetype.name === "Story") {
+        features.push(story);
+      }
+      if (story.fields.issuetype.name === "Bug") {
+        bugs.push(story);
+      }  
+    }
+  }
+}
+
+const jiraProjectKey = process.argv[2];
+const jiraLabel = process.argv[3];
+const slack = process.argv?.[4];
+await collectStories(jiraProjectKey, jiraLabel);
+
+function storyText(story) {
+  const blurbText = extractBlurbText(story.fields.description.content);
```

Comment:
In Jira we could improve this. Instead of looking for a marker in the description like "extraBlurbText" does, we can add a custom blurb field to the issues in Jira. I'm not sure how easy it is to access these custom fields from the jira http api though.

Reply:
@scytacki Good point. If you run `test-jira` the output lists a bunch of `customfield_[n]` fields for the test Jira issue, where`[n]` is some number like 10045. I don't seem to have access to adding/editing custom fields in our Jira account, so I'm not able to thoroughly investigate right now. I _think_ though that, for example, `customfield_10045` corresponds to the "Story Points (Fibonnaci)" field judging by the value and the associated `data-test-id` in the Jira HTML. So if we added a custom "Blurb" field and noted its corresponding ID number, it'd be easy to check for that instead of looking for the marker in the description, like you said.